yarpl/include/yarpl/flowable/FlowableObserveOnOperator.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+namespace yarpl {
+namespace flowable {
+namespace detail {
+
+template <typename T>
+class ObserveOnOperatorSubscriber : public yarpl::flowable::Subscriber<T> {
+ public:
+  ObserveOnOperatorSubscriber(
+      Reference<Subscriber<T>> inner,
+      folly::Executor& executor)
+      : inner_(std::move(inner)), executor_(executor) {}
+
+  // called from upstream EB
+  void onSubscribe(Reference<Subscription> s) override {
+    executor_.add(
+        [ this, self = this->ref_from_this(this), s = std::move(s) ]{
+            inner_->onSubscribe(s);
+        });
+  }
+
+  // called from upstream EB
+  void onNext(T next) override {
+    executor_.add(
+        [ this, self = this->ref_from_this(this), next = std::move(next) ] {
+            inner_->onNext(next);
+        });
+  }
+  void onComplete() override {
+    executor_.add([this, self = this->ref_from_this(this)] {
+        inner_->onComplete();
+    });
+  }
+  void onError(folly::exception_wrapper err) override {
+    executor_.add(
+        [ this, self = this->ref_from_this(this), err = std::move(err) ] {
+            inner_->onError(err);
```

Comment:
Should do `std::move(err)`, and make the lambda `mutable`.  I like naming the two variables differently, so I know if I'm accidentally making a copy where I meant to make a move (i.e. `[e = std::move(err)]() mutable { /* use e */ }`).